src/hooks/index.ts

```diff
@@ -869,3 +893,18 @@ export const useResync = (wallet: YoroiWallet, options?: UseMutationOptions<void
     resync: mutation.mutate,
   }
 }
+
+export const useNfts = (wallet: YoroiWallet) => {
+  const utxos = useUtxos(wallet)
+  const {data, ...rest} = useQuery({
+    queryKey: [wallet.id, 'nfts', utxos.length],
+    queryFn: () => wallet.fetchNfts(),
+  })
+
+  return {...rest, nfts: data ?? []}
+}
+
+export const useNft = (wallet: YoroiWallet, {id}: {id?: string} = {}): YoroiNFT | null => {
+  const {nfts} = useNfts(wallet)
+  return nfts.find((nft) => nft.id === id) ?? null
```

Comment:
what does it mean to not find the nft? does it make sense to throw a "not found" error?

Reply:
I thought this could happen when the cache gets cleared or something similar and the NFTs are not available at that moment. Not really sure if it's better to throw an error and use error boundaries around to handle such case

Reply:
the more we can eliminate `| null | undefined` the better

Reply:
I've now removed the nulls and undefineds and throw error when NFT is not found